ui/lib/components/TimeRangeSelector/index.tsx

```diff
@@ -1,30 +1,48 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useMemo } from 'react'
 import { Dropdown, Button, DatePicker } from 'antd'
 import { ClockCircleOutlined, DownOutlined } from '@ant-design/icons'
 import { getValueFormat } from '@baurine/grafana-value-formats'
 import cx from 'classnames'
 import dayjs from 'dayjs'
+import moment, { Moment } from 'moment'
 import { useTranslation } from 'react-i18next'
 
-import styles from './TimeRangeSelector.module.less'
-import { Moment } from 'moment'
+import styles from './index.module.less'
 
 const { RangePicker } = DatePicker
 
-const RECENT_MINS = [30, 60, 3 * 60, 6 * 60, 12 * 60, 24 * 60]
+const RECENT_SECONDS = [
+  30 * 60,
+  60 * 60,
+  3 * 60 * 60,
+  6 * 60 * 60,
+  12 * 60 * 60,
+  24 * 60 * 60,
+]
 
-export interface TimeRange {
-  recent: number
-  begin_time: number
-  end_time: number
+export interface RecentSecTime {
+  type: 'recent'
+  value: number // unit: seconds
 }
 
-export function getDefTimeRange(): TimeRange {
-  const now = dayjs().unix()
-  return {
-    recent: 30,
-    begin_time: now - 30 * 60,
-    end_time: now,
+export interface RangeTime {
+  type: 'absolute'
+  value: [number, number] // unit: seconds
```

Comment:
Can we use type `TimeRange` from 'antd' here?

Reply:
Does 'antd' have? I don't find it ...

Reply:
In antd 4.0, it changed to type `moment[]`, so this is ok here.